### DISCUSSION TITLE
Is the privacy plugin supported under windows (10) ?

### DISCUSSION THREAD

**Discussion by PlasmaHH** [squidfunk/mkdocs-material#7318] [Q&A] [answered]
Recently I had to activate the privacy plugin and since then a lot of windows users have the following problem:

![mdocs_links](https://github.com/squidfunk/mkdocs-material/assets/1965659/4d3d7514-6c80-4ac1-bce6-0159b3b1f5fd)

mkdocs seems to endlessly try to create a symlink. 

Now symlinks under windows are a strange thing and only supported with the proper group policies or as admin user or when activating developer mode.

Neither is possible with our internal IT for ordinary users, so my question here before going down this is:

Is the privacy plugin even supposed to run under windows or is this expected to fail and only work on linux and similar OSs?

**Comment by alexvoss** [comment]
[This has come up before](https://github.com/squidfunk/mkdocs-material/discussions/6569) and if I understand things correctly this should have been fixed. Looking at the [current source](https://github.com/squidfunk/mkdocs-material-insiders/blob/cc7a2a3428de72d36bdef372fab24de92f954c87/src/plugins/privacy/plugin.py#L496) that produces the warning, it seems the build should succeed but a warning is printed. Can you check if that is the case?

**Reply by squidfunk** [reply]
The comment explains it:

https://github.com/squidfunk/mkdocs-material/blob/fa04d0ca6324282ff11e1033c0743e76ae7f3af4/src/plugins/privacy/plugin.py#L411-L419

__Related__:

- https://github.com/squidfunk/mkdocs-material/issues/4051

**Reply by squidfunk** [reply, marked as answer]
> Is the privacy plugin even supposed to run under windows or is this expected to fail and only work on linux and similar OSs?

Yes, but symlinks need to be createable. The privacy plugin uses symlinks for files without extensions, so there's currently no way around with the current design. However, you do not need the privacy plugin locally, so you can just [disable it for local builds](https://squidfunk.github.io/mkdocs-material/plugins/privacy/#config.enabled).

**Reply by PlasmaHH** [reply]
Ah, I see. I did let it run for longer and it turns out that it doesn't run in an endless loop but for each file, and since there are many and download isn't exactly fast, this took a really really long time...

For enhanced user experience, would it be possible to have a dict/set that stores for which URLs the message was already output? Or maybe even better, do not do the download again?

I will also have a look at selectively activating the privacy plugin only but first need to check if/how we can/have to restrict the build runners to be non-windows only since there seem to be some that don't support symlinks either... Thanks for the suggestion

**Reply by squidfunk** [reply]
We would need to rewrite the plugin to not use symlinks in the case of Windows without adequate permissions, which is not something we're currently optimizing for. The general recommendation is to just disable the plugin. We'll reconsider in the future when we're refactoring the plugin.
